src/lexer/keywords.rs

```diff
@@ -6,91 +6,38 @@ pub struct Keywords {
     pub list: HashMap<String, TokenType>,
 }
 
-impl Keywords {
-    pub fn new() -> Self {
+// Macro definition to decrease lines of code
+macro_rules! keywordize {
+    ($( $words:expr => $func:expr ), *) => {{
         let mut list = HashMap::new();
+        $( for &word in $words { list.insert(word.to_string(), $func); } )*
```

Comment:
I think replacing the pattern as `for &word in &$words` would make the macro more easier to read. Then we can invoke it with using reference operator 
`["",""] => TokenType::*`
instead of 
`&["",""] => TokenType::*`

Reply:
Sure, let me do that